packages/engine/src/scene/systems/ShadowSystem.ts

```diff
@@ -124,36 +146,32 @@ export default async function ShadowSystem(world: World) {
   const shadowComponentQuery = defineQuery([DropShadowComponent, GroupComponent])
 
   const shadowOffset = new Vector3(0, 0.01, 0)
-  const shadowGeometry = new PlaneGeometry(1, 1, 1, 1)
-  const shadowMaterial = new MeshBasicMaterial({
-    side: DoubleSide,
-    transparent: true,
-    depthTest: true,
-    depthWrite: false
-  })
 
-  AssetLoader.loadAsync(`${config.client.fileServer}/projects/default-project/public/drop-shadow.png`).then(
-    (texture: Texture) => {
-      shadowMaterial.map = texture
-      shadowMaterial.needsUpdate = true
-    }
-  )
+  let sceneObjects = Array.from(Engine.instance.currentWorld.objectLayerList[ObjectLayers.Camera] || [])
+
+  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function (props) {
+    if (getShadowsEnabled()) return null
+
+    const dropShadowComponent = getComponent(props.root.entity, DropShadowComponent)
+    const groupComponent = getComponent(props.root.entity, GroupComponent)
+
+    useEffect(() => {
+      world.scene.remove(dropShadows)
+      dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, shadowComponentQuery().length)
+      dropShadows.matrixAutoUpdate = false
+      dropShadows.layers.disable(ObjectLayers.Camera)
+      world.scene.add(dropShadows)
 
-  let dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, 0)
-  dropShadows.matrixAutoUpdate = false
+      const sphere = new Sphere()
+      const minRadius = 0.15
+      new Box3().setFromObject(groupComponent[0]).getBoundingSphere(sphere)
+      dropShadowComponent.radius = Math.max(sphere.radius * 2, minRadius)
+    }, [dropShadowComponent.radius, groupComponent.length, dropShadows])
 
-  const dropShadowReactor = startQueryReactor([DropShadowComponent, GroupComponent], function modifyShadowCount() {
-    world.scene.remove(dropShadows)
-    dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, shadowComponentQuery().length)
-    dropShadows.matrixAutoUpdate = false
-    dropShadows.layers.disable(ObjectLayers.Camera)
-    world.scene.add(dropShadows)
     return null
   })
 
-  let sceneObjects = Array.from(Engine.instance.currentWorld.objectLayerList[ObjectLayers.Camera] || [])
-
-  const CreateDropShadows = () => {
+  const execute = () => {
```

Comment:
```ts
const useShadows = getShadowsEnabled()
if (!useShadows) {
```

This still needs to be the condition for the drop shadow functionality

Reply:
It is in the case of the reactor, I just forgot to reinclude it in execute.